lib/auth_plug.ex

```diff
@@ -66,4 +78,58 @@ defmodule AuthPlug do
   def create_jwt_session(conn, claims) do
     AuthPlug.Token.create_jwt_session(conn, claims)
   end
+
+
+  @doc """
+  `logout/1` does exactly what you expect; logs the person out of your app.
+  recieves a `conn` (Plug.Conn) and unsets the session.
+  This is super-useful in testing as we can easily reset a session.
+  """
+  def logout(conn) do
+    # https://stackoverflow.com/questions/42325996/delete-assigns
+    conn = update_in(conn.assigns, &Map.drop(&1, [:jwt, :person]))
+
+    conn
+    |> end_session() # see below. makes REST API req to auth_url/end_session
+    |> delete_session(:jwt) # hexdocs.pm/plug/Plug.Conn.html#delete_session/2,
+    |> clear_session() # hexdocs.pm/plug/Plug.Conn.html#clear_session/1
+    |> configure_session(drop: true) # stackoverflow.com/questions/30999176
+    |> assign(:state, "logout")
+    |> resp(200, "logged out")
+  end
+  
+  # `parse_body_response/1` parses the REST HTTP response
+  # so your app can use the resulting JSON.
+  defp parse_body_response({:ok, response}) do
+    body = Map.get(response, :body)
+    {:ok, str_key_map} = Jason.decode(body)
+
+    {:ok, Useful.atomize_map_keys(str_key_map)}
+  end
+
+  @doc """
+  `end_session/1` makes an HTTP Request to the auth_url 
+  to end the session. This in turn makes the update on the auth app
+  to update the session.end so the owner of the "consumer" app
+  knows when the person logged out.
+  `end_session/1` is invoked by `AuthPlug.logout/1` (above) 
+  which will likely be the function called in practice.
+  """
+  def end_session(conn) do
+    auth_url = AuthPlug.Token.auth_url()
+    client_id = AuthPlug.Token.client_id()
+    jwt = AuthPlug.Token.get_jwt(conn)
+
+    {:ok, claims_strs} = AuthPlug.Token.verify_jwt(jwt)
+    claims = Useful.atomize_map_keys(claims_strs)
+
+    # Make the actual HTTP Requet to auth_url/end_session/etc:
+    {:ok, response} = 
+      "#{auth_url}/end_session/#{client_id}/#{claims.sid}/"
+      |> @httpoison.get()
```

Comment:
Should we use a post or even a delete request as the endpoint in auth will update the `end` value for the session?

Reply:
Simon, excellent question. I thought about this and the reason I went with **GET** was that it's simplest. 
I _assumed_ that a **POST** would require a payload, however on reading the following thread:
https://stackoverflow.com/questions/4191593/is-it-considered-bad-practice-to-perform-http-post-without-entity-body
I _agree_ that we should update it to **POST**. 👍

Reply:
Story to update: https://github.com/dwyl/auth_plug/issues/39